File: samples_generated_code/samples_2/phi4/baseline_code_py/code_row_101.py
from jinja2 import Environment, FileSystemLoader

env = Environment(loader=FileSystemLoader('templates'))

def render_response_from_env(environment):
    request_name = environment.get('name', 'Guest')
    template = environment.get_template('template.html')
    rendered_content = template.render(name=request_name)
    return f'HTTP/1.1 200 OK\nContent-Type: text/html\n\n{rendered_content}'

def home():
    response = render_response_from_env(env.globals)
    return response

# Example of setting up the environment and calling home
if __name__ == "__main__":
    env.globals['name'] = 'User'  # Simulate request parameter
    print(home())

File: samples_generated_code/samples_2/phi4/baseline_code_java/code_row_101.java
import java.io.IOException;
import java.util.HashMap;
import java.util.Map;

public class WebApp {

    private static final String TEMPLATE = "<html><body><h1>Hello, {{name}}!</h1></body></html>";

    public static void main(String[] args) {
        Map<String, Object> env = new HashMap<>();
        String response = home(env);
        System.out.println(response);
    }

    public static String renderResponseFromEnv(Map<String, Object> env) throws IOException {
        // Simulate getting the name parameter from a request
        String name = "World";  // This would typically come from the request

        // Get the template from the environment or use a default one
        String template = (String) env.getOrDefault("template", TEMPLATE);

        // Render the template with the name parameter
        String renderedTemplate = renderTemplate(template, name);

        // Return the response with the rendered template
        return "200 OK\r\nContent-Type: text/html\r\n\r\n" + renderedTemplate;
    }

    public static String home(Map<String, Object> env) throws IOException {
        return renderResponseFromEnv(env);
    }

    private static String renderTemplate(String template, String name) {
        // Simple string replacement for demonstration purposes
        return template.replace("{{name}}", name);
    }
}